cpp/src/table/table.cpp

```diff
@@ -88,13 +88,27 @@ table_view table::select(std::vector<cudf::size_type> const& column_indices) con
     return table_view(columns);
 }
 
-// Concatenate elements of `tables_to_concat` into a single table_view
-table_view concat(std::vector<table_view> const& tables_to_concat) {
-  std::vector<column_view> concat_cols;
-  for (auto& view : tables_to_concat) {
-    concat_cols.insert(concat_cols.end(), view.begin(), view.end());
+std::unique_ptr<table>
+concatenate(std::vector<table_view> const& tables_to_concat,
+            cudaStream_t stream, rmm::mr::device_memory_resource *mr) {
+  auto tbl = std::make_unique<table>(std::vector<std::unique_ptr<column>>{});
+  if (tables_to_concat.size() == 0) { return tbl; }
```

Comment:
Delete line 94, then use default constructor.

```suggestion
  if (tables_to_concat.size() == 0) { return std::make_unique<table>(); }
```